main.tf

```diff
@@ -45,12 +45,12 @@ resource "aws_route" "public_internet_gateway" {
 # Private routes
 #################
 resource "aws_route_table" "private" {
-  count = "${length(var.azs)}"
+  count = "${var.enable_nat_gateway ? (var.single_nat_gateway ? 1 : length(var.private_subnets)) : 0}"
 
   vpc_id           = "${aws_vpc.this.id}"
   propagating_vgws = ["${var.private_propagating_vgws}"]
 
-  tags = "${merge(var.tags, var.private_route_table_tags, map("Name", format("%s-private-%s", var.name, element(var.azs, count.index))))}"
+  tags = "${merge(var.tags, var.private_route_table_tags, map("Name", (var.single_nat_gateway ? "${var.name}-private" : format("%s-private-%s", var.name, element(var.azs, count.index)))))}"
```

Comment:
I think `tags` can be left as it was since there will be at least one route table element. The only change which is required is to use `var.private_subnets` instead of `var.azs`

Reply:
I was on the fence about this so I wouldn't mind going back if you prefer it the old way.

Reply:
Testing out this patch. I like this code as is. When creating a single nat gateway (and so a single private route table), the route table name tag doesn't need the az.